newton/sim/builder.py

```diff
@@ -3262,6 +3279,69 @@ def add_face(i, j, k):
             except np.linalg.LinAlgError:
                 continue
 
+    def add_contact_sensor(
+        self,
+        sensor_shape: str | list[str] | None = None,
+        sensor_body: str | list[str] | None = None,
+        contact_partners_shape: str | list[str] | None = None,
+        contact_partners_body: str | list[str] | None = None,
+        match_fun: Callable[[str, str], bool] | None = None,
+        include_total: bool = True,
+        prune_noncolliding: bool = False,
+        verbose: bool | None = None,
+    ):
+        """Add a contact sensor view to the model. Return a ContactView.
+        Exactly one of `sensor_shape` or `sensor_body` must be specified to define the sensor. If contact partners
+        are specified, each sensor produces separate readings per contact partner; otherwise, the sensor will read
+        the total contact force.
+
+        Args:
+            sensor_shape: pattern to match sensor shape names; one entity per matching shape.
+            sensor_body: pattern to match sensor body names; one entity per matching body.
+            contact_partners_shape: pattern to match contact partner shape names; one entity per matching shape.
+            contact_partners_body: pattern to match contact partner body names; one entity per matching body.
+            match_fun: function taking a name and a pattern and returning true if the name matches the pattern;
+            defaults to fnmatch. Accepts "re" for `re.match()`.
```

Comment:
I would remove the `"re"` option - the user could just pass some lambda function that uses `re.match()` themselves. The Selection API defaults to `fnmatch` also.